### PR TITLE
Autoretrieve heartbeat endpoint

### PR DESCRIPTION
**Pull request by gmelodie**
Added the autoretrieve heartbeat endpoint. Simple POST endpoint that updates a `LastConnection` field.

### CODE REVIEW

File: util/retrieval.go [hunk 1]
```diff
@@ -15,3 +19,25 @@ type RetrievalProgress struct {
 	Wait   chan struct{}
 	EndErr error
 }
+
+type HeartbeatAutoretrieveResponse struct {
+	Handle         string    `json:"handle"`
+	LastConnection time.Time `json:"lastConnection"`
+}
+
+type AutoretrieveListResponse struct {
```
Comment (alvin-reyes) [inline util/retrieval.go:28]: the `AutoretrieveListResponse` and `InitAutoretrieveResponse` looks to have the same structure. Can we just merge them into `AutoRetrieveResponse` or is there a particular reason for it? e.i is `AutoretrieveListResponse` meant to be a [] list?

File: util/retrieval.go [hunk 2]
```diff
@@ -15,3 +19,25 @@ type RetrievalProgress struct {
...
+	// AddrInfo       *peer.AddrInfo  `json:"addrInfo"`
+	// Address        address.Address `json:"address"`
+	// Hostname       string          `json:"hostname"`
+
+}
+
+type InitAutoretrieveResponse struct {
+	Handle         string `json:"handle"`
+	Token          string `json:"token"`
+	LastConnection time.Time
```
Comment (alvin-reyes) [inline util/retrieval.go:42]: Do we need to add `json:"lastConnection"` on the `LastConnection` property?

**Review by whyrusleeping — APPROVED** [review]
aside from @alvin-reyes's comment, this LGTM

File: util/retrieval.go (hunk 1 above)
Comment (gmelodie) [reply util/retrieval.go:28]: I was following the same structure (read: copy pasting) of the shuttle register and listing api. In the future we might want these two responses to be different.

**Review by alvin-reyes — APPROVED** [review]
LGTM

**Pull request merged**